interfaces/cython/cantera/composite.py

```diff
@@ -617,14 +617,15 @@ def __init__(self, phase, shape=(0,), states=None, extra=None, meta=None):
 
     def __getitem__(self, index):
         states = self._states[index]
+        extra = OrderedDict({key: val[index] for key, val in self._extra.items()})
         if(isinstance(states, list)):
             num_rows = len(states)
             if num_rows == 0:
                 states = None
```

Comment:
Is there a test that exercises this branch? If so, I think it makes sense to see if `extra` needs to be handled in a similar way. On the other hand, I'm not sure how `num_rows` could end up being 0...

To clarify, my concern is that `extra[key]` would have some value associated with it, when it shouldn't.

Reply:
Interesting. It can be empty if you grab multiple elements outside of the defined range. It turns out to do the right thing already, but I added this to the test regardless.